arch/arm/core/cortex_m/pm_s2ram.c

```diff
@@ -17,4 +18,18 @@ __noinit _cpu_context_t _cpu_context;
 /**
  * S2RAM Marker
  */
-__noinit uint32_t marker;
+static __noinit uint32_t marker;
+
+__weak void pm_s2ram_mark_set(void)
+{
+	marker = MAGIC;
+}
+__weak void pm_s2ram_mark_reset(void)
+{
+	marker = 0;
+}
+
+__weak int pm_s2ram_mark_check(void)
```

Comment:
not a fan of non-standard weaks, why not a Kconfig option to signal these need a custom implementation?

Reply:
added Kconfig option.